mitodl/odl-video-service
# Watch Bucket monitoring and processing

#### What are the relevant tickets?
Fixes #101

#### What's this PR do?
Creates a scheduled celery task that will check for the existence of files in an S3 bucket (assigned via settings/environment variable).  Each file found will be copied to the OVS upload bucket, Video and Videofile model objects will be created for it (with `multiangle=True`), and finally the original file will be deleted from the watch bucket.

#### How should this be manually tested?
- Set the following in your .env file:
```
VIDEO_S3_WATCH_BUCKET=odl-video-service-watch-ci
VIDEO_WATCH_BUCKET_FREQUENCY=30
USWITCH_URL=https://s3.amazonaws.com/odl-video-service-uswitch-dev/ci
LECTURE_CAPTURE_USER=<username of existing user>
```
- `docker-compose build`
- `docker-compose up`
- Upload a video file to the `odl-video-service-watch-ci` bucket with an S3 name similar to `MIT-6.046-2017-Spring-lec-mit-0000-2017apr06-0404-L01.mp4`.  Within 30 seconds, the file should start to be processed.
- Once the video has started to be transcoded, the original file should no longer exist in `VIDEO_S3_WATCH_BUCKET`
- The video file should be in the `MIT-6.046-L01-2017-Spring` collection belonging to the user specified in `LECTURE_CAPTURE_USER` - you can either create the collection yourself or it will be created automatically.
- After being transcoded, the USwitch player should load on the video detail page.  You'll need to point any .mit.edu subdomain to your docker host IP for it to work
